diff -u -b -r ../../step5_pi_checkpoint/main.cpp ./main.cpp
--- ../../step5_pi_checkpoint/main.cpp	2016-06-15 14:12:52.874919385 -0400
+++ ./main.cpp	2016-06-16 10:39:58.127693504 -0400
@@ -11,8 +11,7 @@
     alps::params p(argc, (const char**)argv);
 
     // Define the parameters
-    mysim_type::define_parameters(p)
-        .define<std::size_t>("timelimit", 5, "Time limit for the computation");
+    mysim_type::define_parameters(p);
         
     if (p.help_requested(std::cerr) || p.has_missing(std::cerr))
         return 1;

