--- ../../step5_pi_checkpoint/main.cpp	2017-01-27 01:37:14.603122647 -0500
+++ ./main.cpp	2017-01-27 01:53:15.111079851 -0500
@@ -11,8 +11,7 @@
     alps::params p(argc, argv);
 
     // Define the parameters
-    mysim_type::define_parameters(p)
-        .define<std::size_t>("timelimit", 5, "Time limit for the computation");
+    mysim_type::define_parameters(p);
         
     if (p.help_requested(std::cerr) || p.has_missing(std::cerr))
         return 1;
